### PR TITLE
install remote secrets on the clusters

### PR DESCRIPTION
**Pull request by ZhengzheYang**

### CODE REVIEW

File: asmcli/commands/add-to-mesh.sh [hunk 1]
```diff
@@ -89,7 +96,7 @@ add_all_to_mesh() {
         --gke-uri="${GKE_CLUSTER_URI}" \
         --enable-workload-identity
     else
-      { read -r -d '' MSG; fatal "${MSG}"; } <<EOF || true
+      { read -r -d '' MSG; warn "${MSG}"; } <<EOF || true
```
Comment (ZhengzheYang) [inline asmcli/commands/add-to-mesh.sh:99]: this is tentative because the heredoc thing with while loop is coming back. If we are skipping all the clusters (because clusters are already registered to the specified fleet), it will enter the loop nonetheless. Last time we solve this by using for loop, but I am not sure how to read into multiple variables with for loop.

File: asmcli/commands/install.sh [hunk 2]
```diff
@@ -250,3 +250,16 @@ install_control_plane() {
 
   apply_kube_yamls
 }
+
+install_remote_secret() {
```
Comment (zerobfd) [inline asmcli/commands/install.sh:254]: I think this belongs in the `add_to_mesh` file, I don't think it will ever get called in `asmcli install blablabla`

File: asmcli/commands/add-to-mesh.sh [hunk 3]
```diff
@@ -72,15 +75,19 @@ EOF
 
 add_all_to_mesh() {
```
Comment (zerobfd) [inline asmcli/commands/add-to-mesh.sh:76]: Is there a way to refactor this into different functions? It's pretty hard to follow.

**Review by zerobfd — COMMENTED** [review]
Core logic looks right, but we need to break up the complexity

File: asmcli/commands/install.sh (hunk 2 above)
Comment (ZhengzheYang) [reply asmcli/commands/install.sh:254]: I put it here because 1) it has install in its name and 2) we might need it for apply. Should I put this in `util.sh` or just in`add-to-mesh.sh`?

Comment (zerobfd) [reply asmcli/commands/install.sh:254]: For now, in `add-to-mesh`. It might move to `util` later but I'm not 100% sure we'll need it in apply, and I want discrete commits to be consistent.

Comment (ZhengzheYang) [reply asmcli/commands/install.sh]: Done

File: asmcli/commands/add-to-mesh.sh (hunk 3 above)
Comment (ZhengzheYang) [reply asmcli/commands/add-to-mesh.sh:79]: Refactored.

File: asmcli/commands/add-to-mesh.sh (hunk 1 above)
Comment (ZhengzheYang) [reply asmcli/commands/add-to-mesh.sh]: Used while loop with `< <(do_stuff)` structure to avoid blank lines.

**Review by zerobfd — COMMENTED** [review]
Much nicer looking! Can we add one or two unit tests and mock out kubectl/istioctl? (The answer might be no, that's OK.)

**Comment by ZhengzheYang** [comment]
> Much nicer looking! Can we add one or two unit tests and mock out kubectl/istioctl? (The answer might be no, that's OK.)

@zerobfd added one unit test

**Review by zerobfd — APPROVED** [review]
